fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java

```diff
@@ -117,6 +118,11 @@ public Statistics visitCompoundPredicate(CompoundPredicate predicate, Estimation
                     colBuilder.setMinValue(union.getLow()).setMinExpr(union.getLowExpr())
                             .setMaxValue(union.getHigh()).setMaxExpr(union.getHighExpr())
                             .setNdv(union.getDistinctValues());
+                    if (!(leftExpr instanceof IsNull || rightExpr instanceof IsNull)) {
+                        colBuilder.setNumNulls(0);
```

Comment:
A is null or A is not null
=>A.numNulls = 0
this is not true

Reply:
use the max numNulls instead.